### PR TITLE
Change Slider value on MouseWheel event.

### PR DESCRIPTION
**Pull request by xxMUROxx**
Possibility to rotate the mouse wheel and change the value of the slider in following situations:
    * None
    * Only when the slider is focuesed and the mouse wheel is rotated
    * If the mouse pointer is over the slider and the mouse wheel is rotated

### CODE REVIEW

File: MahApps.Metro/Controls/Helper/SliderHelper.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            if ((MouseWheelState)e.NewValue != MouseWheelState.None)
+            {
+                slider.Unloaded += OnUnloaded;
+                slider.PreviewMouseWheel += OnPreviewMouseWheel;
+            }
+            else
+            {
+                UnregisterEvents(slider);
+            }
+        }
```
Comment (punker76) [inline MahApps.Metro/Controls/Helper/SliderHelper.cs]: we should always unregister the events

```csharp
if (e.NewValue != e.OldValue)
{
	var slider = (Slider)d;
	UnregisterEvents(slider);
	if ((MouseWheelState)e.NewValue != MouseWheelState.None)
	{
	    slider.Unloaded += OnUnloaded;
	    slider.PreviewMouseWheel += OnPreviewMouseWheel;
	}
}
```

**Comment by thoemmi** [comment]
BTW, I would introduce `RegisterEvents(Slider slider)`. Though it would be called only from one location, you would get symmetric methods, `RegisterEvents` and `UnregisterEvents`.

**Comment by punker76** [comment]
@thoemmi :+1:

**Comment by xxMUROxx** [comment]
@thoemmi @punker76 Done :+1:

**Comment by khteh** [comment]
Is there any change completion event for this mousewheel change of the slider value?

**Pull request merged**